### PR TITLE
Prays & Blessings

### PR DESCRIPTION
**Pull request by ozcodex**
Adds a new object, the efigy, that allows the player to raise one prayer each day.
this prayer have a probability of result in a blessing from god, that replenish the hunger, energy and thirst.

Future plans include use the character lore tho change the probability of getting a blessing (for example: low if is atheist, higher if is believer), and also make each player to have a different god in which they believe. Im thinking also in include different effects for the different gods, even negative ones (maybe you preffer not to pray if your god is not a nice one)

im open to discussions about this idea. Also I would appreciate receiving a code review, since i have no experience coding mods for minetest, so a lot of mistakes could have been done.

### CODE REVIEW

File: mods/religion/init.lua [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		return is_owner(pos, name) and inv:is_empty("main")
+	end,
+
+	on_rightclick = function(pos, node, player, itemstack, pointed_thing)
+		local meta = player:get_meta()
+		local last_prayer_day = meta:get_int("last_prayer")
+		local current_day = minetest.get_day_count()
+		if( current_day > last_prayer_day) then
+			-- raise a prayer
+			minetest.chat_send_player(player:get_player_name(), S("You raise a prayer to your god"))
```
Comment (jeremyshannon) [inline mods/religion/init.lua:66]: "You raise a prayer to your god" would be better phrased: "you offer up a prayer to your god" in English. And "You have already said your prayers for today" for the prayer time out response.

**Comment by jeremyshannon** [comment]
All in all, this is a neat idea and appears to be written well, but I'm afraid it might have to live as an optional mod, as it clashes with the first point in Dokimi's design philosophy: #23 
>Genre: a hard-scifi core underneath an ambiguous fantasy-sci fi surface. Supernatural elements can exist, but there should be at least the possibility that a scientific explanation exists.

This is basically just straight-up magic. Which is a shame because it's a cool mechanic to invest some time early on carving a statue in exchange for a chance at filling your belly. I just don't see a way to explain it with technology that doesn't sound very hard to swallow.

**Comment by ozcodex** [comment]
Shame on me that forgot about that, but maybe we can change it a bit, making it not a miracle, and more some phsicologycal, remove the part where the stats are replenished, and just leave cool messages and make it more related to the lore.... less mystical/magic and more like a funny mechanic that allow us to define aspects of the personality of the player.

for example: keep a track of how many days the character pray, and if is at least 80% of lived days, give the trait 'devotee' in the biograpy tab, or if is less than 10% give the trait 'atheist'...

and some times the message is "nothing happens", but other times can be "you feel a soft breeze that comforts you"... and have like a tier of messages, and the players can try to pray every day to see if they get a different message.

**Comment by ozcodex** [comment]
By the way this is how the efigy looks
image

File: mods/religion/init.lua (hunk 1 above)
Comment (ozcodex) [reply mods/religion/init.lua]: That sounds great, thanks for the suggestion! as you can guess English is not my first language

File: mods/religion/init.lua [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+local pray_time = 2 
+
+-- the higher the weight, higher is the probability of getting it 
+local pray_responses = {
+	{message="and nothing happens", weight=100},
+	{message="and you feel refreshed", weight=20},
+	{message="you feel how your faith is strengthened", weight=20},
+	{message="something inside you tells you that this is not working", weight=10},
+	{message="and you feel a soft breeze that comforts you", weight=10},
+}
```
Comment (ozcodex) [inline mods/religion/init.lua:19]: Here we can add more messages, and each one have it's own probability (weight) to happen.

**Review by jeremyshannon — CHANGES_REQUESTED** [review]
There should be no need to add the thatch png if you depend on tech, as it already has the texture and will be loaded.
I'm not sure what mods/naturalslopeslib is doing in here, can you revert whatever commit added it as a deleted file?
Otherwise this all looks fine codewise.

**Comment by ozcodex** [comment]
Thank you, i didnt know what happened with naturalslopeslib, sorry for that.
I solve the issues and I think is done!

**Comment by jeremyshannon** [comment]
Code all looks good now!
 Though it occurs to me we could make something like your original idea work, if instead of just restoring your stats directly,  it beams you a sort-of care package of special food that meets all your nutritional requirements, which is manufactured somewhere in the city below.
 The words of the prayer could be a memory of the proper request code for a machine in the city to send you food via a transporter. Though even if you got the words and everything right, the machines don't work very well anymore and may fail to receive your message or fail to make or send the food. I'm unsure how it would hear you though.
 @Dokimi your thoughts on this PR and what I wrote?

**Review by jeremyshannon — APPROVED** [review]

**Comment by DokimiCU** [comment]
I haven't play tested this, so it's hard to judge, but here's some thoughts:
- this is why modding support is important #26. I love the experimentation and ideas. It's inevitable people will come up with stuff that is great content, but doesn't belong in the core game. This might be one of those times... maybe.
- This could be a great custom craft feature limited to specific character backstories #28. In fact, I suspect that's where this best belongs, rather than being accessible to all players at all times.
e.g. your character has a "Priest" or "Shaman" backstory, giving them access. but maybe "Philosopher" or "Scientist" don't get this, they get something else.
- Small energy boost sounds good to me (rather than giving food or water boost - meaning the gods are real). In the sense of a psychological boost.
- Primitive player crafted items maybe shouldn't be able to trigger Artifact technology. e.g. an artifact might deliver teleported food, but not an idol made of sticks (makes more sense to me). Teleportation food idea is an interesting one, but don't see it needing to be "religious". e.g. it would be like worshiping the TV remote control ("Oh mystical power I press thy buttons!").

Overall interesting idea. I think using it as character backstory might be the way to go. We need something to fill out that feature, small weird and amusing things like this are perfect for that. This would add uniqueness to religious characters, and their struggles with their god abandoning them here.

**Comment by DokimiCU** [comment]
Character backstories still needs a framework for how that will be handled, beyond just this idea. I'll leave that one to the skilled programmers. Suggest ideas under #28 .

Inspired by this I've added a suggested principle to Design Philosophy #23 (many ideas like this might best be handled by character backstories)

Also, this idea touches on potentially controversial subjects (e.g. religion), so I've opened a more general discussion #159.

**Pull request closed without merging**